Sources/BaggageContext/BaggageContext.swift

```diff
@@ -0,0 +1,47 @@
+/// `BaggageContextKey`s are used as keys in a `BaggageContext`.
+/// Their associated type `Value` gurantees type-safety.
+public protocol BaggageContextKey {
+    /// The type of `Value` uniquely identified by this key.
+    associatedtype Value
+}
```

Comment:
minot nitpick: let's have BaggageContext be the "first" thing in the file, the other things are secondary (and can be below it)